test/libyul/yulOptimizerTests/fullSSATransform/for_increment.yul

```diff
@@ -0,0 +1,36 @@
+{
+  let a := 0
+  for {} lt(a, 10) {
+    // checked increment
+    if eq(a, sub(0, 1)) { revert(0, 0) }
+    a := add(a, 1)
+  } {
+    sstore(a, 42)
+  }
+}
+// ----
+// step: fullSSATransform
+//
+// {
+//     let a := 0
+//     phi_store("a", a)
+//     for { }
+//     true
+//     {
+//         let a_2 := phi_load("a")
+//         if eq(a_2, sub(0, 1)) { revert(0, 0) }
```

Comment:
I had hoped we could more easily resolve this here, but it won't work just like that, because we'll loose the knowledge from the condition in the body...
Moving the loop post into the body (which I think works and only works for loops without ``continue``) would help - but that'd probably generally help independently of this PR.